src/Yarhl.UnitTests/FileSystem/NodeFactoryTests.cs

```diff
@@ -541,5 +541,35 @@ public void CreateFromSubstreamWithNullStreamThrowsException()
                 () => NodeFactory.FromSubstream("node", null, 0, 0),
                 Throws.ArgumentNullException);
         }
+
+        [Test]
+        public void CreateFromDirectoryWithDoubleSlashes()
+        {
+            // Issue #139
+            string root = Path.GetTempPath();
+            string child = Path.GetRandomFileName();
+            string tempDir = Path.Combine(root, child);
+            Directory.CreateDirectory(tempDir);
+
+            string tempFolder = Path.Combine(tempDir, "folder");
+            Assert.That(Directory.CreateDirectory(tempFolder).Exists, Is.True);
+            string tempFile = Path.Combine(tempFolder, "file.txt");
+            File.Create(tempFile).Dispose();
+
+            Node node = NodeFactory.FromDirectory(
+                string.Concat(root, "//", child),
```

Comment:
Can you change the `//` by `Path.PathSeparator + Path.PathSeparator`, so we test this issue for Linux and Windows (the CI runs in the three platforms)?

Reply:
Done!